### PR TITLE
Refactor seekTranscript to use helper methods.

### PR DESCRIPTION
**Pull request by amuamushu**
Update the structure of `seekTranscript` to use helper methods. The implementation of these helper methods will added in later pull requests.

Edit: Calling the helper methods inside `seekTranscript` would make more sense after the helper methods have been implemented. Thus, this PR will revert to draft until the helpers have been approved.
Edit 2: This PR will close for now because the structure of seekTranscript may change as other pull requests are reviewed so it would be easier to structure this pull request later.

### CODE REVIEW

File: src/main/webapp/view/transcript/transcript.js [hunk 1]
```diff
@@ -112,18 +112,18 @@ function deleteTranscript() {
...
+  if (isBeforeTheFirstTranscriptLine(currentTimeMs)) {
+    resetToBeforeTranscript();
   }
   if (isWithinCurrentTimeRange(currentTimeMs)) {
     addBold(currentTranscriptLine);
     return;
   }
+  const nextTranscript = getNextTranscriptLine(currentTimeMs);
   removeBold(currentTranscriptLine);
-  currentTranscriptLine = currentTranscriptLine.nextElementSibling;
```
Comment (n-wach) [inline src/main/webapp/view/transcript/transcript.js:123]: Can you move this into `getNextTranscriptLine` so we don't lose the seeking functionality we already have?  With a TODO to improve/change ofc

Comment (amuamushu) [reply src/main/webapp/view/transcript/transcript.js:123]: Done, thanks!

File: src/main/webapp/view/transcript/transcript.js [hunk 2]
```diff
@@ -163,6 +163,26 @@ function isWithinCurrentTimeRange(currentTimeMs) {
...
+}
+
+/**
+ * Returns the closest transcript line based on `currentTimeMs`.
+ */
+function getNextTranscriptLine(currentTimeMs) {
+  // TODO: Implement this function to get
+  // the next line based on search.
+  currentTranscriptLine = currentTranscriptLine.nextElementSibling;
+  console.log(currentTimeMs);
```
Comment (jaysaleh) [inline src/main/webapp/view/transcript/transcript.js:182]: Is this needed?

Comment (amuamushu) [reply src/main/webapp/view/transcript/transcript.js:182]: Sort of, I needed to use the parameter for the checks to pass but I'll remove it when I implement the function.

File: src/main/webapp/view/transcript/transcript.js [hunk 3]
```diff
@@ -112,18 +112,18 @@ function deleteTranscript() {
...
-  if (currentTimeMs < currentTranscriptLine.startTimestampMs) {
-    return;
+  if (isBeforeTheFirstTranscriptLine(currentTimeMs)) {
+    resetToBeforeTranscript();
   }
   if (isWithinCurrentTimeRange(currentTimeMs)) {
     addBold(currentTranscriptLine);
     return;
   }
+  const nextTranscript = getNextTranscriptLine(currentTimeMs);
```
Comment (jaysaleh) [inline src/main/webapp/view/transcript/transcript.js:122]: nit: rename to `nextTranscriptLine`

File: src/main/webapp/view/transcript/transcript.js (hunk 2 above)
Comment (jaysaleh) [reply src/main/webapp/view/transcript/transcript.js:182]: Makes sense thanks

File: src/main/webapp/view/transcript/transcript.js [hunk 4]
```diff
@@ -171,3 +191,12 @@ function scrollToTopOfTranscript(transcriptLine) {
   const ulElementOffset = transcriptLine.parentElement.offsetTop;
   transcriptContainer.scrollTop = transcriptLine.offsetTop - ulElementOffset;
 }
+
+/**
+ * Resets the transcript container to what it looks like before it
+ * started scrolling.
+ */
+function resetToBeforeTranscript() {
```
Comment (jaysaleh) [inline src/main/webapp/view/transcript/transcript.js:199]: Is this used right now? It doesn't do anything so maybe add it when you need it

Comment (amuamushu) [reply src/main/webapp/view/transcript/transcript.js:199]: Not really, but I added a function call for this in `seekTranscript` where I would like to call it once it is implemented. I thought of adding a stub as like a TODO. I'll keep it here if that's okay.

File: src/main/webapp/view/transcript/transcript.js (hunk 3 above)
Comment (amuamushu) [reply src/main/webapp/view/transcript/transcript.js]: done! thanks

File: src/main/webapp/view/transcript/transcript.js (hunk 4 above)
Comment (jaysaleh) [reply src/main/webapp/view/transcript/transcript.js:199]: Ah okay

**Pull request closed without merging**